### PR TITLE
storage: add cluster setting delete_compaction_excise.enabled

### PR DESCRIPTION
**Pull request by itsbilal**
This change adds a cluster setting to be able to dynamically
disable excises in delete-only compactions. Excises allow
for range deletions / range key deletions to be applied to
sstables even if they don't fully overlap with the sstable;
we just produce virtual sstables for the remainder of the sstable.

Epic: none

Release note: None

### CODE REVIEW

**Review by aadityasondhi — APPROVED** [review]

File: pkg/storage/pebble.go [hunk 1]
```diff
@@ -111,6 +111,18 @@ var columnarBlocksEnabled = settings.RegisterBoolSetting(
...
+// apply rangedels/rangekeydels on sstables they partially apply to, through
+// an excise operation, instead of just applying the rangedels/rangekeydels
+// that fully delete sstables.
+var deleteCompactionsCanExcise = settings.RegisterBoolSetting(
+	settings.SystemOnly,
+	"storage.delete_compaction_excise.enabled",
+	"set to false to direct Pebble to not partially excise sstables in delete-only compactions",
+	metamorphic.ConstantWithTestBool(
+		"storage.delete_compaction_excise.enabled", true), /* defaultValue */
+	settings.WithPublic)
```
Comment (nicktrav) [inline pkg/storage/pebble.go:124]: Driveby: should this be a public setting?

Comment (itsbilal) [reply pkg/storage/pebble.go:124]: I think there's a good argument for it to go either way. For now I just copied the same visibility as some of the other compaction-y settings like ingest_split.enabled above.a

**Comment by itsbilal** [comment]
TFTR!

bors r=aadityasondhi
